ParseRequestException: i/o failure in ParseDecompressInterceptor on 1.12.0 on API<19 devices

I (and other developers) are getting errors with 1.12.0 on older devices (seemingly API 19 and below). http://stackoverflow.com/questions/34561280/android-parse-sdk-i-o-failure-unknown-format-magic-number-227b

My stacktrace, from an API 18 device:
```
01-19 01:43:23.035 22904-22904/io.givenow.app W/System.err: com.parse.ParseRequest$ParseRequestException: i/o failure
01-19 01:43:23.035 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest.newTemporaryException(ParseRequest.java:289)
01-19 01:43:23.035 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest$2.then(ParseRequest.java:144)
01-19 01:43:23.035 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest$2.then(ParseRequest.java:138)
01-19 01:43:23.035 22904-22904/io.givenow.app W/System.err:     at bolts.Task$15.run(Task.java:839)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at java.lang.Thread.run(Thread.java:841)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err: Caused by: java.io.IOException: unknown format (magic number 227b)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:101)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:81)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseDecompressInterceptor.intercept(ParseDecompressInterceptor.java:40)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseHttpClient$ParseNetworkInterceptorChain.proceed(ParseHttpClient.java:147)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParsePlugins$1.intercept(ParsePlugins.java:115)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseHttpClient$ParseNetworkInterceptorChain.proceed(ParseHttpClient.java:147)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseHttpClient.execute(ParseHttpClient.java:122)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest$3.then(ParseRequest.java:135)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest$3.then(ParseRequest.java:132)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.Task$15.run(Task.java:839)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.BoltsExecutors$ImmediateExecutor.execute(BoltsExecutors.java:105)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.Task.completeAfterTask(Task.java:830)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.Task.continueWithTask(Task.java:642)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.Task.continueWithTask(Task.java:653)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.Task$13.then(Task.java:745)
01-19 01:43:23.045 22904-22904/io.givenow.app W/System.err:     at bolts.Task$13.then(Task.java:733)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err: 	... 4 more
01-19 01:43:23.055 22904-22904/io.givenow.app D/ErrorDialogs: Error caused by: 
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err: java.io.IOException: unknown format (magic number 227b)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:101)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:81)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseDecompressInterceptor.intercept(ParseDecompressInterceptor.java:40)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseHttpClient$ParseNetworkInterceptorChain.proceed(ParseHttpClient.java:147)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParsePlugins$1.intercept(ParsePlugins.java:115)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseHttpClient$ParseNetworkInterceptorChain.proceed(ParseHttpClient.java:147)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseHttpClient.execute(ParseHttpClient.java:122)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest$3.then(ParseRequest.java:135)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at com.parse.ParseRequest$3.then(ParseRequest.java:132)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task$15.run(Task.java:839)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.BoltsExecutors$ImmediateExecutor.execute(BoltsExecutors.java:105)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task.completeAfterTask(Task.java:830)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task.continueWithTask(Task.java:642)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task.continueWithTask(Task.java:653)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task$13.then(Task.java:745)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task$13.then(Task.java:733)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at bolts.Task$15.run(Task.java:839)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
01-19 01:43:23.055 22904-22904/io.givenow.app W/System.err:     at java.lang.Thread.run(Thread.java:841)
```

--- a/Parse/src/main/java/com/parse/Parse.java
+++ b/Parse/src/main/java/com/parse/Parse.java
@@ -387,7 +387,7 @@ public class Parse {
     ParseHttpClient.setKeepAlive(true);
     ParseHttpClient.setMaxConnections(20);
     // If we have interceptors in list, we have to initialize all http clients and add interceptors
-    if (configuration.interceptors != null) {
+    if (configuration.interceptors != null && configuration.interceptors.size() > 0) {
       initializeParseHttpClientsWithParseNetworkInterceptors(configuration.interceptors);
     }
 

